runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/retries/Exceptions.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package aws.smithy.kotlin.runtime.retries
+
+/**
+ * Indicates some failure happened while retrying.
+ * @param message A message indicating the failure mode.
+ * @param cause An underlying exception which caused this exception.
+ * @param attempts The number of attempts made before this failure was encountered.
+ * @param lastResponse The last response received from the retry strategy before this failure was encountered.
+ * @param lastException The exception caught by the retry strategy before this failure was encountered. Note that the
+ * [Throwable] in this parameter isn't necessarily the cause of this exception.
+ */
+sealed class RetryException(
+    message: String,
+    cause: Throwable?,
+    val attempts: Int,
+    val lastResponse: Any?,
+    val lastException: Throwable?,
+) : Exception(message, cause)
```

Comment:
should this inherit from the SDK exception hierarchy? Probably `ClientException` since it's a client side error?